### ISSUE TITLE
text() function in Xpath is case sensitive

### ISSUE THREAD

**Issue by zhinu** [dis-moi/backend#362]
Hi is there a simple way to make this **text()** function not case sensitive?
The workaround I found it to use 

> text()[ contains( translate(., 'ABCDEFGHIJKLMNOPQRSTUVWXYZ', 'abcdefghijklmnopqrstuvwxyz'), 'test')]

instead of 

> text()[contains(.,'test')]

It is pretty annoying, and the input needs to be in lowercase.

**Comment by lutangar** [comment]
Yeah this is problematic,  and yes the workaround is correct and was documented:
https://github.com/dis-moi/backend/blob/master/docs/xpath.md#examples

> See mention of XPATH 2.0 in the same documentation as well

We could use a [Javascript implementation of XPATH 2](https://github.com/ilinsky/xpath2.js) instead of the native xpath 1,  but we'll need to run a few test. As you know, matching engine need to be as fast as possible.

Another option would be, as previously discussed, some kind of *xpath 1 generator*, something friendlier to use.

**Comment by zhinu** [comment]
yep xpath 2.0 with __matches__ would be great!

**Comment by MaartenLMEM** [comment]
Just discussed with Johan. Value of this feature is limited and I better understand downsides mentionned by Johan.
Summary : It impacts just human users. 
It goes back to backlog

**Comment by zhinu** [comment]
Actually xpath 2.0 and 3.0 seem hard to test. Browsers don't natively support them and there aren't many tools available (with a nice looking gui). So yep sticking to 1.0 is probably best for now
